components/accelerator/nvidia/fabric-manager/component.go

```diff
@@ -22,15 +22,9 @@ func New(ctx context.Context, cfg Config) (components.Component, error) {
 	cfg.Query.SetDefaultsIfNotSet()
 
 	cctx, ccancel := context.WithCancel(ctx)
-	nvidia_query.SetDefaultPoller(
-		nvidia_query.WithDBRW(cfg.Log.Query.State.DBRW),
-		nvidia_query.WithDBRO(cfg.Log.Query.State.DBRO),
-		nvidia_query.WithNvidiaSMICommand(cfg.NvidiaSMICommand),
-		nvidia_query.WithNvidiaSMIQueryCommand(cfg.NvidiaSMIQueryCommand),
-		nvidia_query.WithIbstatCommand(cfg.IbstatCommand),
-		nvidia_query.WithInfinibandClassDirectory(cfg.InfinibandClassDirectory),
-	)
-	nvidia_query.GetDefaultPoller().Start(cctx, cfg.Query, Name)
+	if nvidia_query.GetDefaultPoller() != nil {
```

Comment:
why do we need to check nil here but not in other components?

Reply:
for example here https://github.com/leptonai/gpud/pull/341/files#diff-c050a270659092b2aa3a80d35443598a080348e5283b5cab50d35414774656cbL30